### ISSUE TITLE
WebFluxLinkBuilder does not work when ReactorDebugAgent is active.

### ISSUE THREAD

**Issue by ctytgat** [spring-projects/spring-hateoas#1144]
Using spring-hateoas 1.0.1

ReactorDebugAgent is automatically enabled when having the reactor-tools lib on the classpath.
This adds extra instrumentation around Mono instances etc.

Using linkTo() will fail with the stacktrace below.
The Mono is not of the expected proxy type, but a MonoOnAssembly added by ReactorDebugAgent.

```
java.lang.IllegalArgumentException: Object of class [reactor.core.publisher.MonoOnAssembly] must be an instance of interface org.springframework.hateoas.server.core.LastInvocationAware
	at org.springframework.util.Assert.instanceCheckFailed(Assert.java:696)
...
Stack trace:
		at org.springframework.util.Assert.instanceCheckFailed(Assert.java:696)
		at org.springframework.util.Assert.isInstanceOf(Assert.java:596)
		at org.springframework.util.Assert.isInstanceOf(Assert.java:627)
		at org.springframework.hateoas.server.core.WebHandler.linkTo(WebHandler.java:90)
		at org.springframework.hateoas.server.core.WebHandler.linkTo(WebHandler.java:83)
		at org.springframework.hateoas.server.reactive.WebFluxLinkBuilder.linkToInternal(WebFluxLinkBuilder.java:250)
		at org.springframework.hateoas.server.reactive.WebFluxLinkBuilder.linkToInternal(WebFluxLinkBuilder.java:243)
		at org.springframework.hateoas.server.reactive.WebFluxLinkBuilder.linkTo(WebFluxLinkBuilder.java:69)
```

**Comment by bsideup** [comment]
ok, I was able to reproduce it with `HypermediaWebFilterTest`.

The proxy of `TestController` will look like this:
```java
Mono<T> root() {
   return proxyCallTo("root");
}
```
where `proxyCallTo` is the cglib’s magic that intercepts the call, applies the provided logic, etc etc

The agent will instrument it (since there are no operator invocations in it), so that `methodOn(TestController.class).root()` will call this method instead:
```java
Mono<T> root() {
   return addAssemblyInfo(proxyCallTo("root"), ...);
}
```

where `addAssemblyInfo` is what `ReactorDebugAgent` adds (and changes to `MonoOnAssembly`)

I tried to make the agent ignore synthetic classes, but apparently cglib does not mark generated classes as synthetic.

@odrotbohm is there a way to mark them as synthetic when you create a proxy?
